### ISSUE TITLE
Uninformative error message for ambiguous parse

### ISSUE THREAD

**Issue by nad** [agda/agda#3135]
Let's say that we have the following code:
```agda
postulate
  □   : Set
  _□  : Set → Set
  ⟨_⟩ : Set → Set
```
In this setting the following ambiguous code yields a somewhat informative error message:
```agda
G : Set
G = ? □
```
```
Don't know how to parse ? □. Could mean any one of:
  ? □
  _□ ?
Operators used in the grammar:
  □ (postfix operator, level 20) [_□ ([...])]
when scope checking ? □
```
However, if the following minor variation of the code above is used, then the error message is less informative:
```agda
G : Set
G = ⟨ ? □ ⟩
```
```
Don't know how to parse ⟨ ? □ ⟩. Could mean any one of:
  ⟨ (? □) ⟩
  ⟨ (? □) ⟩
Operators used in the grammar:
  □   (postfix operator, level 20) [_□ ([...])] 
  ⟨_⟩ (closed operator)            [⟨_⟩ ([...])]
when scope checking ⟨ ? □ ⟩
```
I suppose that it would be easy to make the latter error message match the former one.

**Comment by andreasabel** [comment]
@nad, it would be probably easy to print the second error as:
```
Could mean any one of:
  ⟨_⟩ (? □)
  ⟨_⟩ (_□ ?)
```
Just switch off operator reconstruction.  Not sure this is what you want.  So what do you want?

**Comment by nad** [comment]
What is "operator reconstruction"?

In cases where two distinct expressions are printed in the same way I think it makes sense to find some way to explain how they differ. I'm not sure if it is better to print the expressions in the way you suggest, or in the following way:
```
  ⟨ ? □ ⟩
  ⟨ _□ ? ⟩
```

**Comment by pnlph** [comment]
> In cases where two distinct expressions are printed in the same way I think it makes sense to find some way to explain how they differ.

I agree. Don't know if it's related, but I tried with a similar code
```agda
postulate
  A : Set
  □   : Set
  _□  : Set → Set
  ⟨_⟩ : Set → Set

open import Agda.Builtin.Equality using (_≡_ ; refl)
_ : A □ ≡ A □
_ = refl
```
and when loading the file, I got:
```
Don't know how to parse A □ ≡ A □. Could mean any one of:
  (A □) ≡ (A □)
  (A □) ≡ (A □)
  (A □) ≡ (A □)
  (A □) ≡ (A □)
```
At a glance, I could not find any difference among the 4 options, but Agda does somehow.

**Comment by andreasabel** [comment]
> What is "operator reconstruction"?

This is a phase in `AbstractToConcrete` responsible for printing `1 + 2` instead of just `_+_ 1 2`.  But anyway, this is not relevant here because we are still on the level of `Concrete` syntax.

> In cases where two distinct expressions are printed in the same way I think it makes sense to find some way to explain how they differ.

One could just unparse operator expressions in the parts that differ.   (To keep the outermost, shared operator forms in place.)
